engine/src/main/java/org/hibernate/validator/internal/engine/ValidatorImpl.java

```diff
@@ -167,17 +167,19 @@ public ValidatorImpl(ConstraintValidatorFactory constraintValidatorFactory,
 	public final <T> Set<ConstraintViolation<T>> validate(T object, Class<?>... groups) {
 		Contracts.assertNotNull( object, MESSAGES.validatedObjectMustNotBeNull() );
 
-		if ( !beanMetaDataManager.isConstrained( object.getClass() ) ) {
```

Comment:
The issue is not `isConstrained` by itself, that's why I kept it and there's still a usage of it.

The issue was that we used `beanMetaDataManager.isConstrained` which got the bean metadata from the concurrent map and then we got again the bean metadata from the concurrent map a few lines below (either really a few lines below or hidden in a method call a few lines below). We should get the bean metadata from the concurrent map only once.